tests/ui/diverging_sub_expression.rs

```diff
@@ -35,6 +37,18 @@ fn foobar() {
                 99 => return,
                 _ => true || panic!("boo"),
             },
+            // lint blocks as well
+            15 => true || { return; },
+            16 => false || { return; },
+            // ... and when it's a single expression
+            17 => true || { return },
+            18 => false || { return },
+            // ... but not when there's both an expression and a statement
+            19 => true || { _ = 1; return },
+            20 => false || { _ = 1; return },
```

Comment:
Just noticed this. The comment above says this shouldn't lint, but it currently does. One of the two should be fixed.